File: presences/src/main/resources/public/ts/models/Audience.ts
import {Mix} from 'entcore-toolkit';
import http from 'axios';

export class Audience {
    name: string;
    id: string;
    type: string;

    constructor(name: string, id?, type?) {
        this.name = name;
        if (id)
            this.id = id;
        else if (type)
            this.type = type;
    }

    toString(): string {
        return this.name;
    }
}

export class Audiences {
    all: Audience[];

    constructor() {
        this.all = [];
    }

    /**
     * Synchronize groups belongs to the parameter structure
     * @param structureId structure id
     * @returns {Promise<void>}
     */
    async sync(structureId: string) {
        try {
            let audiences = await http.get('/viescolaire/classes?idEtablissement=' + structureId);
            this.all = Mix.castArrayAs(Audience, audiences.data);
        } catch (e) {
            throw e;
        }
    }
}

File: statistics-presences/src/main/java/fr/openent/statistics_presences/bean/Audience.java
package fr.openent.statistics_presences.bean;

public class Audience {
    private String id;
    private String name;

    public Audience(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String id() {
        return this.id;
    }

    public String name() {
        return this.name;
    }
}
